monai/transforms/compose.py

```diff
@@ -318,15 +318,15 @@ def __len__(self):
         """Return number of transformations."""
         return len(self.flatten().transforms)
 
-    def __call__(self, input_, start=0, end=None, threading=False, lazy: bool | None = None):
+    def __call__(self, input_, start=0, end=None, threading=False, lazy: bool | None = False):
         result = execute_compose(
             input_,
             transforms=self.transforms,
             start=start,
             end=end,
             map_items=self.map_items,
             unpack_items=self.unpack_items,
-            lazy=self.lazy,
+            lazy=lazy,
```

Comment:
thanks for the PR! The expected logic here is that if `lazy is None`, then `self.lazy` will be used.

Reply:
Okay, I will incorporate your suggestion, @wyli.

Reply:
There are testing errors, please follow the existing code logic so that it's consistent in API usage, for example https://github.com/Project-MONAI/MONAI/blob/a86c0e01555727bd848275c7eabc5b9dc26da73d/monai/transforms/spatial/array.py#L223